cmd/stcrashreceiver/stcrashreceiver.go

```diff
@@ -78,14 +82,42 @@ func (r *crashReceiver) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// serveGet responds to GET requests by serving the uncompressed report.
+func (r *crashReceiver) serveGet(base string, w http.ResponseWriter, req *http.Request) {
+	path := filepath.Join(r.dirFor(base), base)
+	fullPath := filepath.Join(r.dir, path)
+	if fd, err := os.Open(fullPath + ".gz"); err == nil {
+		// There is a compressed report
+		defer fd.Close()
+		gr, err := gzip.NewReader(fd)
+		if err != nil {
+			http.Error(w, "Internal server error", http.StatusInternalServerError)
+			return
+		}
+		_, _ = io.Copy(w, gr) // best effort
+		return
+	}
+	if _, err := os.Lstat(fullPath); err == nil {
+		http.ServeFile(w, req, fullPath)
+		return
+	}
+	http.Error(w, "Not found", http.StatusNotFound)
+}
+
 // serveHead responds to HEAD requests by checking if the named report
 // already exists in the system.
 func (r *crashReceiver) serveHead(base string, w http.ResponseWriter, _ *http.Request) {
 	path := filepath.Join(r.dirFor(base), base)
-	if _, err := os.Lstat(path); err != nil {
-		http.Error(w, "Not found", http.StatusNotFound)
+	fullPath := filepath.Join(r.dir, path)
+	if _, err := os.Lstat(fullPath + ".gz"); err == nil {
+		// There is a compressed report
+		return
 	}
-	// 200 OK
+	if _, err := os.Lstat(fullPath); err == nil {
```

Comment:
It looks like `path` vs `fullpath` is a change in behaviour unrelated to compression. As far as I understand before just `path` was used, but `fullPath` was used to store reports, so it shouldn't have worked/worked just by accident. Now `path` doesn't seem to be used at all except for creating `fullPath`, so could be removed. I feel like I must be missing something here.

Reply:
It didn't work before, unless the dir happened to be `"."` which it was when I was testing. (On the other hand it didn't matter much as it only affected the HEAD request to see if the report was already uploaded, and when uploaded it was also marked as done on the client side.)

Reply:
Ok, thanks for the clarification. You could drop `path` here and in `serveGet` then and construct `fullPath` in a single call to `filepath.Join` to prevent anyone (me) searching for the place where `path` is used :)